api/v1/client.go

```diff
@@ -1209,3 +1214,104 @@ func (c *storageClient) Validator(ctx context.Context, r *http.Request) (*Valida
 
 	return &v, nil
 }
+
+// TransactionsPerSecond returns a list of tps checkpoint values.
+func (c *storageClient) TransactionsPerSecond(ctx context.Context, r *http.Request) (*TpsCheckpointList, error) {
+	qf := NewQueryFactory(strcase.ToSnake(LatestChainID))
+
+	pagination := common.Pagination{
+		Limit:  100,
+		Offset: 0,
+	}
+
+	rows, err := c.db.Query(
+		ctx,
+		qf.TpsCheckpointQuery(),
+		pagination.Limit,
+		pagination.Offset,
+	)
+	if err != nil {
+		c.logger.Info("query failed",
+			"request_id", ctx.Value(RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrStorageError
+	}
+	defer rows.Close()
+
+	ts := TpsCheckpointList{
+		IntervalMinutes: tpsWindowSizeMinutes,
+		TpsCheckpoints:  []TpsCheckpoint{},
+	}
+	for rows.Next() {
+		var d struct {
+			Hour     time.Time
+			MinSlot  int
+			TxVolume uint64
+		}
+		if err := rows.Scan(
+			&d.Hour,
+			&d.MinSlot,
+			&d.TxVolume,
+		); err != nil {
+			c.logger.Info("query failed",
+				"err", err.Error(),
+			)
+			return nil, common.ErrStorageError
+		}
+
+		t := TpsCheckpoint{
+			Timestamp: d.Hour.Add(time.Duration(tpsWindowSizeMinutes*d.MinSlot) * time.Minute),
+			TxVolume:  d.TxVolume,
+		}
+		ts.TpsCheckpoints = append(ts.TpsCheckpoints, t)
+	}
+
+	return &ts, nil
+}
+
+// DailyVolumes returns a list of daily transaction volumes.
+func (c *storageClient) DailyVolumes(ctx context.Context, r *http.Request) (*VolumeList, error) {
+	qf := NewQueryFactory(strcase.ToSnake(LatestChainID))
+
+	pagination := common.Pagination{
```

Comment:
In the same vein, are users able to query tx volumes in the past before the prior 100 days? Seems like it might be useful for block explorers?

Reply:
Hm. Fair enough. I'd assumed 100 was okay since block explorers typically just show a graph of recent tx volume, but you're right -- no harm in just letting them query further back if needed.